radixdlt/babylon-wallet-android
# [Abw-2454] Verify user written seed phrase

## Description
This PR adds verification step after user confirm he/she backed up her seed phrase on Reveal Seed phrase screen. User is asked to fill in random 4 words of mnemonic, and those words are compared with corresponding words from mnemonic stored on device.


## How to test

1. Create account and transfer some XRD into it, for a seed phrase that was not yet backed up.
2. Tap prompt to back up mnemonic on account card or account details and reveal seed phrase.
3. When closing, confirm you written down the phrase.
4. Pass additional verification.

## Video

https://github.com/radixdlt/babylon-wallet-android/assets/118203440/9b73871b-cf98-4901-b1fe-f633eebd7b8d



## PR submission checklist
- [x] I have tested seed phrase verification and seed phrase input in backup restoration
